modules/monitoring-modifying-retention-time-and-size-for-prometheus-metrics-data.adoc

```diff
@@ -0,0 +1,135 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: PROCEDURE
+[id="modifying-retention-time-and-size-for-prometheus-metrics-data_{context}"]
+= Modifying the retention time and size for Prometheus metrics data
+
+By default, Prometheus automatically retains metrics data for 15 days.
+You can modify the retention time to change how soon data is deleted by specifying a time value in the `retention` field.
+You can also configure the maximum amount of disk space used by the retained metrics data by specifying a size value in the `retentionSize` field.
+If this size limit is reached, the oldest data is deleted first until the disk space used by the data is again below the maximum.
+
+Note the following behaviors of these data retention settings:
+
+* The retention settings apply only to data block directories. 
+The settings are not used for data located in the `wal` and `chunks_head` directories.
+* The retention settings are used only when Prometheus writes a new data block to a data block directory, which occurs every two hours and after the write-ahead log (WAL) has recorded at least three hours of data.
```

Comment:
Line 16 to 18 would need to rephrase:

Size-based retention takes into account only the block directories in `/prometheus`.  In other words it doesn't include the `wal` and `chunks_head`  directories.
Also the retention policy is only applied when Prometheus cuts a new block which happens every 2 hours (and after the WAL contains at least 3h of data).

Reply:
@slashpai - I updated the wording in lines 16-18 to read:

```
* The size-based retention policy applies only to data block directories in the `/prometheus` directory. 
This policy does not apply to data located in the `/wal` and `/chunks_head` directories.
* The size-based retention policy is applied only when Prometheus cuts a new data block, which occurs every two hours and after the write-ahead log (WAL) contains at least three hours of data.
```
Please review.

Reply:
I think this could still be improved here. Based on https://bugzilla.redhat.com/show_bug.cgi?id=2078755#c11 and whats noted in https://github.com/prometheus/prometheus/blob/main/docs/storage.md#operational-aspects I think we should highlight two aspects for size-based retention:

1. `wal` and `head_chunks` will count towards the size limit. So a limit lower than the `max wal size + head chunks` is effectively not keeping any data blocks.
2. A size limit (or a time limit for that matter) will never delete any `wal` or `head_chunk` data. This data does count towards the size limit, but is never affected by it.

Point 2 is well addressed here I think. Maybe 1 is not quite as explicit here as it could be? This can cause confusion, see the bug linked above.

cc @simonpasquier for visibilty.

Reply:
@jan--f @simonpasquier Per Jan's suggestion, I've revised the content as follows:

```
* The size-based retention policy applies only to data block directories in the `/prometheus` directory. 
This policy does not apply to data located in the `/wal` and `/chunks_head` directories.
* Data in the `/wal` and `/head_chunks` directories counts toward the retention size limit, even though data from these directories will never be purged based on size- or time-based retention policies. 
Thus, if you set a retention size limit lower than the maximum size set for the `/wal` and `/head_chunks` directories, in effect you have configured the system not to retain any data blocks in the data block directories.
```

Reply:
Better change as
* The size-based retention policy applies all data bytes in the `/prometheus` directory, including persistent blocks, WAL and m-mapped chunks. 
* Data in the `/wal` and `/head_chunks` directories counts toward the retention size limit, even though data from these directories will never be purged based on size- or time-based retention policies. 
Thus, if you set a retention size limit lower than the maximum size set for the `/wal` and `/head_chunks` directories, in effect you have configured the system not to retain any data blocks in the prometheus data directories.

Reply:
Updated per your suggestion @lihongyan1